src/Dialogs/Settings/Panels/WeatherConfigPanel.cpp

```diff
@@ -49,13 +58,31 @@ class WeatherConfigPanel final
   /* methods from Widget */
   virtual void Prepare(ContainerWindow &parent, const PixelRect &rc) override;
   virtual bool Save(bool &changed) override;
+
+  private:
+  /* methods from DataFieldListener */
+  virtual void OnModified(DataField &df) override {};
 };
 
+static void
+FillRegionControl(WndProperty &wp, const TCHAR *setting)
+{
+  DataFieldEnum *df = (DataFieldEnum *)wp.GetDataField();
+  auto skysight = DataGlobals::GetSkysight();
+
+  for (auto &i: skysight->GetRegions())
+    df->addEnumText(i.first.c_str(), i.second.c_str());
+
+  // if old region doesn't exist any more this will fall back to first element
+  df->Set(setting);
+  wp.RefreshDisplay();
+}
+
 void
 WeatherConfigPanel::Prepare(ContainerWindow &parent, const PixelRect &rc)
 {
   const auto &settings = CommonInterface::GetComputerSettings().weather;
-
+  WndProperty *wp;
```

Comment:
Move variable declaration to where it's used

Reply:
Move to the one it is assigned.